### ISSUE TITLE
resilience4j plugin rule config data bug

### ISSUE THREAD

**Issue by Redick01** [dromara/soul#1091]
## Describe the bug

- Question one :

soul-admin: if you do not fill in the resilience4j plugin rule config data, request proxy interface will report an error，The error content is convert json exception, the code is as follows:
```
Resilience4JHandle resilience4JHandle = GsonUtils.getGson().fromJson(rule.getHandle(), Resilience4JHandle.class); 
```
- Question two : 

soul-admin: the args "degrade opening duration" must be more than the 1000 otherwise , will report an error ,the error is "Illegal argument interval: 0 millis is less than 1", the throw exception code is "Duration.ofSeconds(handle.getWaitIntervalFunctionInOpenState() / 1000" in the Resilience4JBuilder.java 57 line

## Environment

- Soul version(s): [e.g. 2.2.1]

## Debug logs
- Question one Exception Stack trace

```
com.google.gson.JsonSyntaxException: java.lang.NumberFormatException: empty String
	at com.google.gson.internal.bind.TypeAdapters$7.read(TypeAdapters.java:228) ~[gson-2.8.6.jar:na]
	Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException: 
Error has been observed at the following site(s):
	|_ checkpoint ⇢ org.dromara.soul.web.configuration.ErrorHandlerConfiguration$1 [DefaultWebFilterChain]
	|_ checkpoint ⇢ org.dromara.soul.web.filter.WebSocketParamFilter [DefaultWebFilterChain]
	|_ checkpoint ⇢ org.springframework.boot.actuate.metrics.web.reactive.server.MetricsWebFilter [DefaultWebFilterChain]
	|_ checkpoint ⇢ HTTP POST "/soul/test/hello?req=1111" [ExceptionHandlingWebHandler]
Stack trace:
		at com.google.gson.internal.bind.TypeAdapters$7.read(TypeAdapters.java:228) ~[gson-2.8.6.jar:na]
		at com.google.gson.internal.bind.TypeAdapters$7.read(TypeAdapters.java:218) ~[gson-2.8.6.jar:na]
		at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:131) ~[gson-2.8.6.jar:na]
		at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:222) ~[gson-2.8.6.jar:na]
		at com.google.gson.Gson.fromJson(Gson.java:932) ~[gson-2.8.6.jar:na]
		at com.google.gson.Gson.fromJson(Gson.java:897) ~[gson-2.8.6.jar:na]
		at com.google.gson.Gson.fromJson(Gson.java:846) ~[gson-2.8.6.jar:na]
		at com.google.gson.Gson.fromJson(Gson.java:817) ~[gson-2.8.6.jar:na]
		at org.dromara.soul.plugin.resilience4j.Resilience4JPlugin.doExecute(Resilience4JPlugin.java:62) ~[soul-plugin-resilience4j-2.2.1.jar:2.2.1]
		at org.dromara.soul.plugin.base.AbstractSoulPlugin.execute(AbstractSoulPlugin.java:97) ~[soul-plugin-base-2.2.1.jar:2.2.1]
		at org.dromara.soul.web.handler.SoulWebHandler$DefaultSoulPluginChain.lambda$execute$0(SoulWebHandler.java:107) ~[soul-web-2.2.1.jar:2.2.1]
		at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:44) [reactor-core-3.3.1.RELEASE.jar:3.3.1.RELEASE]
		at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:52) [reactor-core-3.3.1.RELEASE.jar:3.3.1.RELEASE]
		at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:52) [reactor-core-3.3.1.RELEASE.jar:3.3.1.RELEASE]
		at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:52) [reactor-core-3.3.1.RELEASE.jar:3.3.1.RELEASE]
		at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:52) [reactor-core-3.3.1.RELEASE.jar:3.3.1.RELEASE]
		at reactor.core.publisher.Mono.subscribe(Mono.java:4105) ~[reactor-core-3.3.1.RELEASE.jar:3.3.1.RELEASE]
		at reactor.core.publisher.MonoSubscribeOn$SubscribeOnSubscriber.run(MonoSubscribeOn.java:124) ~[reactor-core-3.3.1.RELEASE.jar:3.3.1.RELEASE]
		at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:84) ~[reactor-core-3.3.1.RELEASE.jar:3.3.1.RELEASE]
		at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:37) ~[reactor-core-3.3.1.RELEASE.jar:3.3.1.RELEASE]
		at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[na:1.8.0_231]
		at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) ~[na:1.8.0_231]
		at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) ~[na:1.8.0_231]
		at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[na:1.8.0_231]
		at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[na:1.8.0_231]
		at java.lang.Thread.run(Thread.java:748) ~[na:1.8.0_231]
```

- Question two Exception Stack trace

```
java.lang.IllegalArgumentException: Illegal argument interval: 0 millis is less than 1
	at io.github.resilience4j.core.IntervalFunctionCompanion.checkInterval(IntervalFunction.java:251) ~[resilience4j-core-1.6.1.jar:1.6.1]
	Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException: 
Error has been observed at the following site(s):
	|_ checkpoint ⇢ org.dromara.soul.web.configuration.ErrorHandlerConfiguration$1 [DefaultWebFilterChain]
	|_ checkpoint ⇢ org.dromara.soul.web.filter.WebSocketParamFilter [DefaultWebFilterChain]
	|_ checkpoint ⇢ org.springframework.boot.actuate.metrics.web.reactive.server.MetricsWebFilter [DefaultWebFilterChain]
	|_ checkpoint ⇢ HTTP POST "/soul/test/hello?req=1111" [ExceptionHandlingWebHandler]
Stack trace:
		at io.github.resilience4j.core.IntervalFunctionCompanion.checkInterval(IntervalFunction.java:251) ~[resilience4j-core-1.6.1.jar:1.6.1]
		at io.github.resilience4j.core.IntervalFunction.of(IntervalFunction.java:54) ~[resilience4j-core-1.6.1.jar:1.6.1]
		at io.github.resilience4j.core.IntervalFunction.of(IntervalFunction.java:68) ~[resilience4j-core-1.6.1.jar:1.6.1]
		at org.dromara.soul.plugin.resilience4j.build.Resilience4JBuilder.build(Resilience4JBuilder.java:57) ~[soul-plugin-resilience4j-2.2.1.jar:2.2.1]
		at org.dromara.soul.plugin.resilience4j.Resilience4JPlugin.combined(Resilience4JPlugin.java:76) ~[soul-plugin-resilience4j-2.2.1.jar:2.2.1]
		at org.dromara.soul.plugin.resilience4j.Resilience4JPlugin.doExecute(Resilience4JPlugin.java:64) ~[soul-plugin-resilience4j-2.2.1.jar:2.2.1]
		at org.dromara.soul.plugin.base.AbstractSoulPlugin.execute(AbstractSoulPlugin.java:97) ~[soul-plugin-base-2.2.1.jar:2.2.1]
		at org.dromara.soul.web.handler.SoulWebHandler$DefaultSoulPluginChain.lambda$execute$0(SoulWebHandler.java:107) ~[soul-web-2.2.1.jar:2.2.1]
		at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:44) [reactor-core-3.3.1.RELEASE.jar:3.3.1.RELEASE]
		at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:52) [reactor-core-3.3.1.RELEASE.jar:3.3.1.RELEASE]
		at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:52) [reactor-core-3.3.1.RELEASE.jar:3.3.1.RELEASE]
		at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:52) [reactor-core-3.3.1.RELEASE.jar:3.3.1.RELEASE]
		at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:52) [reactor-core-3.3.1.RELEASE.jar:3.3.1.RELEASE]
		at reactor.core.publisher.Mono.subscribe(Mono.java:4105) ~[reactor-core-3.3.1.RELEASE.jar:3.3.1.RELEASE]
		at reactor.core.publisher.MonoSubscribeOn$SubscribeOnSubscriber.run(MonoSubscribeOn.java:124) ~[reactor-core-3.3.1.RELEASE.jar:3.3.1.RELEASE]
		at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:84) ~[reactor-core-3.3.1.RELEASE.jar:3.3.1.RELEASE]
		at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:37) ~[reactor-core-3.3.1.RELEASE.jar:3.3.1.RELEASE]
		at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[na:1.8.0_231]
		at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) ~[na:1.8.0_231]
		at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) ~[na:1.8.0_231]
		at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[na:1.8.0_231]
		at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[na:1.8.0_231]
		at java.lang.Thread.run(Thread.java:748) ~[na:1.8.0_231]
```

<details>
<summary>Click to expand</summary>

```
replace this line with your debug logs
```
</details>

**Comment by nuo-promise** [comment]
> ## Describe the bug
> * Question one :
> 
> soul-admin: if you do not fill in the resilience4j plugin rule config data, request proxy interface will report an error，The error content is convert json exception, the code is as follows:
> 
> ```
> Resilience4JHandle resilience4JHandle = GsonUtils.getGson().fromJson(rule.getHandle(), Resilience4JHandle.class); 
> ```
> 
> * Question two :
> 
> soul-admin: the args "degrade opening duration" must be more than the 1000 otherwise , will report an error ,the error is "Illegal argument interval: 0 millis is less than 1", the throw exception code is "Duration.ofSeconds(handle.getWaitIntervalFunctionInOpenState() / 1000" in the Resilience4JBuilder.java 57 line
> 
> ## Environment
> * Soul version(s): [e.g. 2.2.1]
> 
> ## Steps to reproduce
> ## Expected behavior
> ## Debug logs
> * Question one Exception Stack trace
> 
> ```
> com.google.gson.JsonSyntaxException: java.lang.NumberFormatException: empty String
> 	at com.google.gson.internal.bind.TypeAdapters$7.read(TypeAdapters.java:228) ~[gson-2.8.6.jar:na]
> 	Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException: 
> Error has been observed at the following site(s):
> 	|_ checkpoint ⇢ org.dromara.soul.web.configuration.ErrorHandlerConfiguration$1 [DefaultWebFilterChain]
> 	|_ checkpoint ⇢ org.dromara.soul.web.filter.WebSocketParamFilter [DefaultWebFilterChain]
> 	|_ checkpoint ⇢ org.springframework.boot.actuate.metrics.web.reactive.server.MetricsWebFilter [DefaultWebFilterChain]
> 	|_ checkpoint ⇢ HTTP POST "/soul/test/hello?req=1111" [ExceptionHandlingWebHandler]
> Stack trace:
> 		at com.google.gson.internal.bind.TypeAdapters$7.read(TypeAdapters.java:228) ~[gson-2.8.6.jar:na]
> 		at com.google.gson.internal.bind.TypeAdapters$7.read(TypeAdapters.java:218) ~[gson-2.8.6.jar:na]
> 		at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:131) ~[gson-2.8.6.jar:na]
> 		at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:222) ~[gson-2.8.6.jar:na]
> 		at com.google.gson.Gson.fromJson(Gson.java:932) ~[gson-2.8.6.jar:na]
> 		at com.google.gson.Gson.fromJson(Gson.java:897) ~[gson-2.8.6.jar:na]
> 		at com.google.gson.Gson.fromJson(Gson.java:846) ~[gson-2.8.6.jar:na]
> 		at com.google.gson.Gson.fromJson(Gson.java:817) ~[gson-2.8.6.jar:na]
> 		at org.dromara.soul.plugin.resilience4j.Resilience4JPlugin.doExecute(Resilience4JPlugin.java:62) ~[soul-plugin-resilience4j-2.2.1.jar:2.2.1]
> 		at org.dromara.soul.plugin.base.AbstractSoulPlugin.execute(AbstractSoulPlugin.java:97) ~[soul-plugin-base-2.2.1.jar:2.2.1]
> 		at org.dromara.soul.web.handler.SoulWebHandler$DefaultSoulPluginChain.lambda$execute$0(SoulWebHandler.java:107) ~[soul-web-2.2.1.jar:2.2.1]
> 		at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:44) [reactor-core-3.3.1.RELEASE.jar:3.3.1.RELEASE]
> 		at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:52) [reactor-core-3.3.1.RELEASE.jar:3.3.1.RELEASE]
> 		at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:52) [reactor-core-3.3.1.RELEASE.jar:3.3.1.RELEASE]
> 		at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:52) [reactor-core-3.3.1.RELEASE.jar:3.3.1.RELEASE]
> 		at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:52) [reactor-core-3.3.1.RELEASE.jar:3.3.1.RELEASE]
> 		at reactor.core.publisher.Mono.subscribe(Mono.java:4105) ~[reactor-core-3.3.1.RELEASE.jar:3.3.1.RELEASE]
> 		at reactor.core.publisher.MonoSubscribeOn$SubscribeOnSubscriber.run(MonoSubscribeOn.java:124) ~[reactor-core-3.3.1.RELEASE.jar:3.3.1.RELEASE]
> 		at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:84) ~[reactor-core-3.3.1.RELEASE.jar:3.3.1.RELEASE]
> 		at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:37) ~[reactor-core-3.3.1.RELEASE.jar:3.3.1.RELEASE]
> 		at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[na:1.8.0_231]
> 		at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) ~[na:1.8.0_231]
> 		at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) ~[na:1.8.0_231]
> 		at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[na:1.8.0_231]
> 		at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[na:1.8.0_231]
> 		at java.lang.Thread.run(Thread.java:748) ~[na:1.8.0_231]
> ```
> 
> * Question two Exception Stack trace
> 
> ```
> java.lang.IllegalArgumentException: Illegal argument interval: 0 millis is less than 1
> 	at io.github.resilience4j.core.IntervalFunctionCompanion.checkInterval(IntervalFunction.java:251) ~[resilience4j-core-1.6.1.jar:1.6.1]
> 	Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException: 
> Error has been observed at the following site(s):
> 	|_ checkpoint ⇢ org.dromara.soul.web.configuration.ErrorHandlerConfiguration$1 [DefaultWebFilterChain]
> 	|_ checkpoint ⇢ org.dromara.soul.web.filter.WebSocketParamFilter [DefaultWebFilterChain]
> 	|_ checkpoint ⇢ org.springframework.boot.actuate.metrics.web.reactive.server.MetricsWebFilter [DefaultWebFilterChain]
> 	|_ checkpoint ⇢ HTTP POST "/soul/test/hello?req=1111" [ExceptionHandlingWebHandler]
> Stack trace:
> 		at io.github.resilience4j.core.IntervalFunctionCompanion.checkInterval(IntervalFunction.java:251) ~[resilience4j-core-1.6.1.jar:1.6.1]
> 		at io.github.resilience4j.core.IntervalFunction.of(IntervalFunction.java:54) ~[resilience4j-core-1.6.1.jar:1.6.1]
> 		at io.github.resilience4j.core.IntervalFunction.of(IntervalFunction.java:68) ~[resilience4j-core-1.6.1.jar:1.6.1]
> 		at org.dromara.soul.plugin.resilience4j.build.Resilience4JBuilder.build(Resilience4JBuilder.java:57) ~[soul-plugin-resilience4j-2.2.1.jar:2.2.1]
> 		at org.dromara.soul.plugin.resilience4j.Resilience4JPlugin.combined(Resilience4JPlugin.java:76) ~[soul-plugin-resilience4j-2.2.1.jar:2.2.1]
> 		at org.dromara.soul.plugin.resilience4j.Resilience4JPlugin.doExecute(Resilience4JPlugin.java:64) ~[soul-plugin-resilience4j-2.2.1.jar:2.2.1]
> 		at org.dromara.soul.plugin.base.AbstractSoulPlugin.execute(AbstractSoulPlugin.java:97) ~[soul-plugin-base-2.2.1.jar:2.2.1]
> 		at org.dromara.soul.web.handler.SoulWebHandler$DefaultSoulPluginChain.lambda$execute$0(SoulWebHandler.java:107) ~[soul-web-2.2.1.jar:2.2.1]
> 		at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:44) [reactor-core-3.3.1.RELEASE.jar:3.3.1.RELEASE]
> 		at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:52) [reactor-core-3.3.1.RELEASE.jar:3.3.1.RELEASE]
> 		at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:52) [reactor-core-3.3.1.RELEASE.jar:3.3.1.RELEASE]
> 		at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:52) [reactor-core-3.3.1.RELEASE.jar:3.3.1.RELEASE]
> 		at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:52) [reactor-core-3.3.1.RELEASE.jar:3.3.1.RELEASE]
> 		at reactor.core.publisher.Mono.subscribe(Mono.java:4105) ~[reactor-core-3.3.1.RELEASE.jar:3.3.1.RELEASE]
> 		at reactor.core.publisher.MonoSubscribeOn$SubscribeOnSubscriber.run(MonoSubscribeOn.java:124) ~[reactor-core-3.3.1.RELEASE.jar:3.3.1.RELEASE]
> 		at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:84) ~[reactor-core-3.3.1.RELEASE.jar:3.3.1.RELEASE]
> 		at reactor.core.scheduler.WorkerTask.call(WorkerTask.java:37) ~[reactor-core-3.3.1.RELEASE.jar:3.3.1.RELEASE]
> 		at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[na:1.8.0_231]
> 		at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) ~[na:1.8.0_231]
> 		at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) ~[na:1.8.0_231]
> 		at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[na:1.8.0_231]
> 		at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[na:1.8.0_231]
> 		at java.lang.Thread.run(Thread.java:748) ~[na:1.8.0_231]
> ```
> 
> Click to expand

fix #1093

**Comment by Redick01** [comment]
fix #1093
